app/models/game.rb

```diff
@@ -107,4 +107,101 @@ def populate_game
       pieces.create(x_position: 6, y_position: y, piece_number: 11, type: 'Pawn')
     end
   end
+
+  def can_claim_draw?(current_user)
+    return true if stalemate?(current_user)
+    return true if threefold_repetition?
+    return true if fifty_move_rule?
+    return false
+  end
+
+  def stalemate?(current_user)
```

Comment:
This method is very long and doesn't provide much detail on what it is actually doing. Maybe we could break it up into smaller methods with descriptive names so that it is clear what it is checking for?